### PR TITLE
Enhance full-stack test

### PR DESCRIPTION
**Pull request by okurz**
* Improve test stack reporting by avoiding Test::More tests in helper libs
* t: Extract common full stack function "schedule_one_job_over_api_and_verify"
* t: Add timeout for loops in full-stack.t
* t: Speedup full stack test with KVM when available

### CODE REVIEW

**Comment by okurz** [comment]
Unfortunately so far I was unable to execute a full stack test successfully locally so I don't know how the test results will actually look like.

File: t/full-stack.t [hunk 1]
```diff
@@ -105,16 +105,11 @@ $driver->title_is("openQA", "back on main page");
...
-  . 'QEMU_NO_FDC_SET=1 CDMODEL=ide-cd HDDMODEL=ide-drive VERSION=1 TEST=core PUBLISH_HDD_1=core-hdd.qcow2 '
-  . 'UEFI_PFLASH_VARS=/usr/share/qemu/ovmf-x86_64.bin';
-
-subtest 'schedule job' => sub {
-    OpenQA::Test::FullstackUtils::client_call("jobs post $JOB_SETUP");
-    OpenQA::Test::FullstackUtils::verify_one_job_displayed_as_scheduled($driver);
-};
+# speedup using virtualization support if available, results should be
+# equivalent, just saving some time
+$OpenQA::Test::FullstackUtils::JOB_SETUP .= ' QEMU_NO_KVM=1' unless -r '/dev/kvm';
```
Comment (Martchus) [inline t/full-stack.t:110]: This line would make sense for `t/33-developer_mode.t` as well so you could move it to the utils.

Comment (okurz) [reply t/full-stack.t]: done

**Comment by okurz** [comment]
https://app.circleci.com/pipelines/github/os-autoinst/openQA/2421/workflows/926b970e-40b2-483d-af23-fafef890029c/jobs/22775/parallel-runs/0/steps/0-108 failed in t/ui/01-list.t with

```
#   Failed test 'only softfailed and failed jobs displayed'

#   at t/ui/01-list.t line 342.

#     Structures begin differing at:

#          $got->[5] = Does not exist

#     $expected->[5] = 'job_99944'

[18:29:53] t/ui/01-list.t ............................. 59/? 

#   Failed test '99945 again hidden'

#   at t/ui/01-list.t line 355.

#     Structures begin differing at:

#          $got->[8] = 'job_99945'
```

but the relevant code part is already covered by `wait_for_ajax`. Could it be we also need to wait for animations here? I don't see it. I will retrigger this test.

**Comment by Martchus** [comment]
The test code looks like this:

```
$driver->get('/tests?resultfilter=Failed&foo=bar&resultfilter=Softfailed');
$driver->find_element_by_id('relevantfilter')->click();
wait_for_ajax();
@jobs = map { $_->get_attribute('id') } @{$driver->find_elements('#results tbody tr', 'css')};
is_deeply(
    \@jobs,
    [qw(job_99940 job_99939 job_99938 job_99936 job_99962 job_99944)],
    'only softfailed and failed jobs displayed'
);
```

After clicking the filter checkbox (`relevantfilter`) there's indeed an AJAX query. The data is displayed by via DataTables. Usually we don't have additional handling for that and I also don't see any animations. So I'm not sure what's causing this. Maybe DataTables postpones rendering at some point so the page stays responsive?

**Comment by okurz** [comment]
Ok, how to check if this is true? Can we use the browser developer tools to find out?

**Comment by okurz** [comment]
@Martchus btw, as you already looked. Wanna approve or ask for changes?

**Comment by okurz** [comment]
@cdywan as you previously looked into test failure stack level changes this should be interesting to you

**Comment by okurz** [comment]
@kalikiana as you previously looked into test failure stack level changes this should be interesting to you

**Comment by okurz** [comment]
failing in t/api/14-plugin_obs_rsync_async.t unrelated to my changes.

**Pull request merged**